tools/autodeployment/webhook.js

```diff
@@ -32,10 +32,18 @@ router.post('/', createNodeMiddleware(webhooks, { path: '/' }));
  * @param {string} repo - Repository name (should be 'telescope')
  * @param {string} buildType - Build type: staging or production
  * @param {string} action - Action for a specific release event: created, published or released.
+ * @param {string} ref - .Git reference resource.
+ * @param {string} mainBranch - Brach used in the repo as main branch.
  */
-function requestFilter(repo, buildType, action) {
+function requestFilter(repo, buildType, action, ref, mainBranch) {
+  const currentBranch = ref.split('/').pop();
+
   return (
+    // Check repo
     repo === REPO_NAME &&
+    // Check if the reference is the branch used as main or if it is a tag
+    (currentBranch === mainBranch || ref.includes('tags/')) &&
```

Comment:
Should we pass `ref_type` instead, which would be one of `branch` or `tag`, so we don't have to rely on string manipulation to determine this?

Reply:
I'm afraid [push](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push) doesn't include `ref_type` in its payload. I think only [create](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#create) and [delete](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#delete) have it.

Reply:
That seems odd.  Thanks for checking.